src/game/ServerList.cpp

```diff
@@ -1,40 +1,64 @@
 #include "ServerList.h"
 
-#include "utils/json.hpp"
+#include <optional>
 
-#include <format>
+#include "utils/JsonValidation.h"
+#include "utils/Log.h"
+#include "utils/json.hpp"
 
 ServerList::ServerList(const std::string& serverlist_json) {
-    auto respobj = nlohmann::json::parse(serverlist_json);
-
-    // todo: better validation!!! important!!
-    for (auto serverobj : respobj) {
-        ServerEntry entry;
-
-        entry.name = serverobj["name"];
-        entry.description = serverobj["description"];
-        entry.hostname = serverobj["ip"];
+    nlohmann::basic_json respobj;
+    try {
+        respobj = nlohmann::json::parse(serverlist_json);
+    }
+    catch (nlohmann::json::exception e) {
+        Log::log_print(FATAL, e.what());
```

Comment:
rethrow instead of logging fatal. the fatal log function actually crashes the software rn due to not properly stopping running threads. no, this is not your fault. we need a better and more consistent approach to exception handling across the codebase. for now, just rethrow.